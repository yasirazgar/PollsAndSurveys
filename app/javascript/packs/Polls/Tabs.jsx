import React, { Component } from 'react';

import Button from '../Utils/Button'
import Polls from './Polls'
import UserPolls from './UserPolls'
import RespondedPolls from './RespondedPolls'
import {pollsTxt, myPollsTxt, respondedPollsTxt} from '../constants.js'

class Tabs extends Component {
  constructor(props) {
    super(props);

    this.pollsClass = 'tablinks active'
    this.myPollsClass = 'tablinks'
    this.respondedPollsClass = 'tablinks'

    this.state = {
      tab: pollsTxt,
    };
  }

  pollsHandler = () => {
    this.pollsClass = 'tablinks active'
    this.myPollsClass = 'tablinks'
    this.respondedPollsClass = 'tablinks'
    this.props.callback(pollsTxt, <Polls />)
  }

  myPollsHandler = () => {
    this.pollsClass = 'tablinks'
    this.myPollsClass = 'tablinks active'
    this.respondedPollsClass = 'tablinks'
    this.props.callback(myPollsTxt, <UserPolls />)
  }

  respondedPollsHandler = () => {
    this.pollsClass = 'tablinks'
    this.myPollsClass = 'tablinks'
    this.respondedPollsClass = 'tablinks active'
    this.props.callback(respondedPollsTxt, <RespondedPolls />)
  }

  render() {
    let tabs = [<Button classes={this.pollsClass} clickHandler={this.pollsHandler} text={pollsTxt} />]

    if (this.props.user){
      tabs.push(<Button classes={this.myPollsClass} clickHandler={this.myPollsHandler} text={myPollsTxt} />)
      tabs.push(<Button classes={this.respondedPollsClass} clickHandler={this.respondedPollsHandler} text={respondedPollsTxt} />)
    }

    return (
     <div className="tab">
       {tabs}
     </div>
    );
  }
}

export default Tabs
