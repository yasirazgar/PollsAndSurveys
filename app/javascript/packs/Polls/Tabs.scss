.tabs {
  border: 1px solid #ccc;
  background-color: #f1f1f1;
  display: flex;
  overflow: hidden;
  position: sticky;
  top: 18%;
  width: 100%;
  z-index: 10;

  /* Style the buttons inside the tab */
  button {
    background-color: inherit;
    float: left;
    border: none;
    outline: none;
    cursor: pointer;
    padding: 14px 16px;
    transition: 0.3s;
    font-size: 17px;

    /* Change background color of buttons on hover */
    &:hover {
      background-color: #ddd;
    }

    /* Create an active/current tablink class */
    &.active {
      background-color: #ccc;
    }
    &.btn__inner {
      float: right;
    }
  }


  /* Style the tab content */
  .tabcontent {
    display: none;
    padding: 6px 12px;
    border: 1px solid #ccc;
    border-top: none;
  }
}