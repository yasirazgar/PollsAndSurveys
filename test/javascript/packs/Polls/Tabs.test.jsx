import React from 'react'
import { shallow } from 'enzyme'
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";

import Tabs from 'packs/Polls/Tabs'

const mockStore = configureMockStore();

describe("Tabs", () => {
  it("renders correctly when logged out", () => {
    const wrapper = shallow(
      <Provider store={mockStore({})} >
        <Tabs />
      </Provider>
    )

    expect(wrapper).toMatchSnapshot();
  });

  it("renders correctly when logged in", () => {
    const wrapper = shallow(
      <Provider store={mockStore({})}>
        <Tabs user={{name: 'user'}}/>
      </Provider>
    )

    expect(wrapper).toMatchSnapshot();
  });
})
