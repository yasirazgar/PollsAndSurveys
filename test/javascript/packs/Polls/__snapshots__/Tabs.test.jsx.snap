// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs renders correctly when logged in 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      },
    }
  }
>
  <Connect(Tabs)
    user={
      Object {
        "name": "user",
      }
    }
  />
</ContextProvider>
`;

exports[`Tabs renders correctly when logged out 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      },
    }
  }
>
  <Connect(Tabs) />
</ContextProvider>
`;
